### ISSUE TITLE
Sections features

### ISSUE THREAD

**Issue by alessandrotraversi** [rainlab/blog-plugin#290]
I tried to subdivide the content into many chapters extending this plugin.

I have created the plugin acme.section, the model and the table, but I don't know how adding a create and remove buttons to a new tab...

Can I create a new tab "section" and where is the possibility of creating/deleting more than one section, when creating or editing a post?

Thank you so much

**Comment by LukeTowers** [comment]
I'm not sure what exactly you are asking

**Comment by alessandrotraversi** [comment]
sorry... my fault!

I have created a new plugin in my project Acme.Blog. In this plugin, I have this fields: id, post_id, title, content and classes.

In my model (Acme.Blog.Models) I have :

```
class Section extends Model
{
    (...)
    public $belongsTo = ['post' => ['RainLab\Blog\Models\Post'],];
    (...)
}
```

and in my Plugin.php:
```
(...)
use RainLab\Blog\Models\Post as PostModel;
use RainLab\Blog\Controllers\Posts as PostsController;

    (...)
    public function boot()
    {
        PostModel::extend(function($model) {
            $model->hasMany['sections'] = ['Acme\Blog\Models\Section'];
        });

        PostsController::extendFormFields(function($widget) {
            if (!$widget->model instanceof PostModel) {
                return;
            }

            $configFile = __DIR__ . '/config/post_fields.yaml';
            $config = Yaml::parse(File::get($configFile));
            $widget->addSecondaryTabFields($config);
        });
    }
    (...)

```
I need to create a tab in view of creation / update of posts but, in addition to exhibiting new fields, it shuld be able to create more than one "section" (records) for each post.
Like the first view of this url "http://myoctober/backend/rainlab/blog/categories" (default), but inside a tab of view post.

Thanks for your help

**Comment by LukeTowers** [comment]
To "add" a tab, you should just be able to specify a `tab: myNewTab` property for each field that you want to add to the overall fields. If you want to add to an existing tab, then just make sure that every field has `tab:` set to the exact name (probably in the form of a lang key, i.e. `rainlab.blog::lang.tabs.whatever`) of the tab that you want to add that field to.

Does that answer your question?

**Comment by alessandrotraversi** [comment]
unfortunately no... create the new tab it isn't the problem for me

please, look this image:

![Interface on new tab](https://atemporale.it/storage/app/media/Senza-titolo-1.jpg)

I'd like an interface like the framed in red. This partial should be placed on my new tab, like the next img:

![Interface on new tab](https://atemporale.it/storage/app/media/Senza-titolo-2.jpg)

This should be the final result...

Basically, my difficulty is to extend the Posts controller. I need to show a corect inteface able to create a series of records (that belong to a new custom table).

Thanks for your time

**Comment by LukeTowers** [comment]
So when you do 
```php
$widget->addSecondaryTabFields($field);
```
with the `tab` property of your new fields set to `newtab` that doesn't add it to the interface?

**Comment by alessandrotraversi** [comment]
I know how add a new tab but I don't understand how add the feature "create" and "delete"

![Interface on new tab](https://atemporale.it/storage/app/media/Senza-titolo-3.jpg)
(this is a photomontage to explain the final result)

**Comment by alessandrotraversi** [comment]
any ideas?

**Comment by LukeTowers** [comment]
Ah, your question didn't make sense to me at first.

You're going to want to implement the RelationController behavior on the PostsController, and then setup the relationConfig property on the PostsController as well: https://octobercms.com/forum/post/extend-blog-plugin-to-implement-relationcontroller
http://octobercms.com/docs/backend/relations

Give that a try and then let me know if you run into the same issue as the first guy in that forum post did.

**Comment by alessandrotraversi** [comment]
Yes. This was exactly what I wanted to know / understand.

... but unfortunately, it is returning this error:

```
Class RainLab\Blog\Controllers\Posts must define property $relationConfig used by Backend\Behaviors\RelationController behavior.

(...)\modules\backend\classes\ControllerBehavior.php line 53
```
I had already tried this solution before, but it doesn't works. I'm sure that there is something wrong or missing in my configuration, but I don't understand what...

I have created a new repository to share with you:
https://github.com/alessandrotraversi/blog

Thanks for your help

**Comment by LukeTowers** [comment]
Alright you should be able to extend the Posts controller now with the most recent commit: https://github.com/rainlab/blog-plugin/commit/ea9db64d7f8c57f66d538008ea354408a62940b2

You may have to apply the change in that commit manually until the plugin gets updated through the marketplace.

**Comment by alessandrotraversi** [comment]
thank you very much, 

this has solved the problem of the error message but it still doesn't work. 
It dosen't show a new tab and if I try to extend an existing tab (for example "edit_tab") ... nothing changes.

What am I doing wrong?

**Comment by LukeTowers** [comment]
Why did you set your Author name of your plugin to `Acme`? That's just a demo name, you should be using your own author name instead.

**Comment by alessandrotraversi** [comment]
Hehe... yes yes I know. The reason is that we want to strongly focus on OctoberCMS, creating an area dedicated to the development of themes and plugins for the community and convincing our clients to migrate from WP to OctoberCMS. I also considered to inscribe in yours partnership program. Unfortunately, we do not know what name to use (yet), so for now, is fine any name.

**Comment by alessandrotraversi** [comment]
have you find any error or missing?

**Comment by LukeTowers** [comment]
Haven't looked yet, I'm quite busy unfortunately. I'll try to take another look at some point.

**Comment by LukeTowers** [comment]
Closing as it has been over a month since any activity on this occurred.
